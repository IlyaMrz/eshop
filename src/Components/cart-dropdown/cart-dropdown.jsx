import React from 'react';
import CartItem from '../cart-item/cart-item';
import CustomButton from '../custom-button/custom-button';
import './cart-dropdown.scss';
import { connect } from 'react-redux';
import { selectCartItems } from '../../redux/cart/cart.selectors';

const CartDowndrop = ({cartItems}) => (
    <div className='cart-dropdown'>
        <div className='cart-items'>
            {
                cartItems.map(cartItem => <CartItem key={cartItem.id} item={cartItem}/>)
            }
            <CustomButton>GO TO CHECKOUT</CustomButton>
        </div>
    </div>
)
const mapStateToProps = (state) => ({
    cartItems: selectCartItems(state)
})

export default connect(mapStateToProps, null)(CartDowndrop);