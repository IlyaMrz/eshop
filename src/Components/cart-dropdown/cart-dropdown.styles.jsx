import styled from 'styled-components';
import CustomButton from '../custom-button/custom-button';

export const CartDropdownButton = styled(CustomButton)`
  margin-top: auto;
`;

export const CartDowndropContainer = styled.div`
    position: absolute;
    width: 240px;
    min-height: 340px;
    max-height: 48vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 20px;
    border: 1px solid black;
    background-color: white;
    top: 90px;
    right: 40px;
    z-index: 5;
    @media screen and (max-width: 400px ) {
        max-height: 70vh;
    }
`;

CartDropdownButton.displayName = 'CartDropdownButton';

export const CartItemsContainer = styled.div`
    width: 90%
    display: flex;
    flex-direction: column;
    overflow: scroll;
    overflow-x: hidden; 
    margin-bottom: 10px;

`;


export const EmptyMessageContainer = styled.span`
font-size: 18px;
margin: 50px auto;
`;

EmptyMessageContainer.displayName = 'EmptyMessageContainer';