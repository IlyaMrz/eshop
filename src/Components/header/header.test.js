import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './header';
import { CartDowndrop } from '../cart-dropdown/cart-dropdown';

describe('Header component', () => {
  let wrapper;
  let mockSignOutStart;

  beforeEach(() => {
    mockSignOutStart = jest.fn();

    const mockProps = {
      hidden: true,
      currentUser: {
        uid: '123'
      },
      signOutStart1: mockSignOutStart
    };

    wrapper = shallow(<Header {...mockProps} />);
  });

  it('should render Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('if currentUser is present', () => {
    it('should render sign out link', () => {
      expect(
        wrapper
          .find('OptionLink')
          .at(3)
          .text()
      ).toBe('SIGN OUT');
    });

    it('should call signOutStart1 method when link is clicked', () => {
      wrapper
        .find('OptionLink')
        .at(3)
        .simulate('click');

      expect(mockSignOutStart).toHaveBeenCalled();
    });
  });

  describe('if currentUser is null', () => {
    it('should render sign in link', () => {
      const mockProps = {
        hidden: true,
        currentUser: null,
        signOutStart1: mockSignOutStart
      };

      const newWrapper = shallow(<Header {...mockProps} />);

      expect(
        newWrapper
          .find('OptionLink')
          .at(3)
          .text()
      ).toBe('SIGN IN');
    });
  });

  describe('if hidden is true', () => {
    it('should not render CartDowndrop', () => {
      expect(wrapper.exists(CartDowndrop)).toBe(false);
    });
  });

  describe('if currentUser is null 2', () => {
    it('should render CartDowndrop', () => {
      const mockProps = {
        hidden: false,
        currentUser: null,
        signOutStart1: mockSignOutStart
      };

      const newWrapper = shallow(<Header {...mockProps} />);
      expect(newWrapper.find('withRouter(Connect(CartDowndrop))').exists()).toBe(true);
    });
  });
})