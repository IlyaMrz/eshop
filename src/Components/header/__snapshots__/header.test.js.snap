// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    <ForwardRef(SvgCrown)
      className="logo"
    />
  </Styled(Link)>
  <styled.div>
    <OptionLink
      to="/"
    >
      HOME
    </OptionLink>
    <OptionLink
      to="/shop"
    >
      SHOP
    </OptionLink>
    <OptionLink
      to="/contact"
    >
      CONTACT
    </OptionLink>
    <OptionLink
      as="div"
      onClick={[MockFunction]}
    >
      SIGN OUT
    </OptionLink>
    <Connect(CartIcon) />
  </styled.div>
</styled.div>
`;
